packages/graphql-languageservice/src/http/HttpLanguageService.ts

```diff
@@ -0,0 +1,67 @@
+import { LanguageService, GraphQLLspConfig } from 'LanguageService';
+import { createHttpSchemaLoader } from './schema-loader';
+import { createHttpQueryExecutor } from './query-executor';
+
+type EndpointConfig = { uri: string; requestOpts?: RequestInit };
+type SeparateIntrospectionAndExecutionConfig = {
+  introspection: EndpointConfig;
+  execution?: EndpointConfig;
+};
+
+type UnifiedConfig = EndpointConfig & { disableExecution?: boolean };
+
+export type HttpGraphQLLanguageServiceConfig = Omit<
+  GraphQLLspConfig,
+  'schemaLoader' | 'queryExecutor'
+> &
+  (UnifiedConfig | SeparateIntrospectionAndExecutionConfig);
+
+function createSchemaLoaderFromHttpGraphQLLanguageServiceConfig(
+  config: HttpGraphQLLanguageServiceConfig,
+) {
+  let endpointConfig: undefined | EndpointConfig = undefined;
+  if ((config as UnifiedConfig).uri) {
+    endpointConfig = config as UnifiedConfig;
+  } else {
+    endpointConfig = (config as SeparateIntrospectionAndExecutionConfig)
+      .introspection;
+  }
+  return createHttpSchemaLoader({
+    introspectionOptions: config.introspectionOptions,
+    uri: endpointConfig.uri,
+    requestOpts: endpointConfig.requestOpts,
+  });
+}
+
+function createQueryExecutorFromHttpGraphQLLanguageServiceConfig(
+  config: HttpGraphQLLanguageServiceConfig,
+) {
+  let endpointConfig: undefined | EndpointConfig = undefined;
+  if ((config as EndpointConfig).uri) {
+    const unified = config as UnifiedConfig;
+    endpointConfig = unified.disableExecution ? undefined : unified;
+  } else {
+    endpointConfig = (config as SeparateIntrospectionAndExecutionConfig)
+      .execution;
+  }
+  if (endpointConfig) {
+    return createHttpQueryExecutor({
+      uri: endpointConfig.uri,
+      requestOpts: endpointConfig.requestOpts,
+    });
+  }
+}
+
+export class HttpFetcherLanguageService extends LanguageService {
```

Comment:
huh. not a huge fan of extending classes at all. also, we should only have a single language service class

Reply:
We could probably then move it back, just with more sophisticated typings of the config? @acao

Reply:
sure, though I'm not sure what further sophistication is needed, it's hard to see what you are referring to?